### PR TITLE
Send all statistics to telemetry consumers

### PR DESCRIPTION
**Pull request by jason-bragg**
Only a hard coded set of statistics were going to telemetry consumers.  Now all non-string statistics are tracked.

### CODE REVIEW

**Comment by jason-bragg** [comment]
tagging @galvesribeiro, as he's more familiar with these systems than I.

File: src/Orleans/Statistics/CounterStatistic.cs [hunk 1]
```diff
@@ -33,6 +33,16 @@ public interface ICounter
...
         string GetDisplayString();
         CounterStorage Storage { get; }
+        void TrackMetric(Logger logger);
+    }
+
+    public static class Metric
+    {
+        public static string CreateName(ICounter counter)
+        {
+            return counter.Name + "." + (counter.IsValueDelta ? "Delta" : "Current");
```
Comment (galvesribeiro) [inline src/Orleans/Statistics/CounterStatistic.cs]: I wonder if we could cache those names on first use...

File: src/Orleans/Statistics/CounterStatistic.cs [hunk 2]
```diff
@@ -141,7 +151,7 @@ private static CounterStatistic FindOrCreate_Impl(StatisticName name, bool useDe
             }
         }
 
-        static public bool Delete(string name)
+        public static bool Delete(string name)
```
Comment (xiazen) [inline src/Orleans/Statistics/CounterStatistic.cs]: the same question here. Just a little confused about why you turn a lot of private method into public. A little explanation will help

**Comment by galvesribeiro** [comment]
For some reason my review was deleted by GitHub o.O

Anyway, the only comment I have is regarding the `Metric.Create()`... I was saying that I wonder if we couldn't cache the metric name instead of concat strings each time it is tracked....

Besides that, LGTM, I should did that before. 

Thanks for doing it @jason-bragg 👍

**Comment by xiazen** [comment]
@galvesribeiro I think I deleted them by accident. I was deleting my false comments...

but anyway, LGTM. I will let @jason-bragg  to decide whether to fix the Metric.Create() or not

**Comment by galvesribeiro** [comment]
@jason-bragg thanks for the latest changes.

Regarding the delta, I always think that deltas and other calculations should be managed by the ones consuming/processing the metric.

LGTM :shipit: !

**Comment by xiazen** [comment]
@galvesribeiro I second you opinion on delta, Orleans should just be responsible for outputting raw data. All the post processing should be managed by the ones consume the data.

**Comment by galvesribeiro** [comment]
@xiazen yup! That was the major principle of bring those APIs for Telemetry. Lets hope at some point everything telemetry-related is flowing that way.

**Pull request merged**